src/snowflake/snowpark/_internal/compiler/query_generator.py

```diff
@@ -121,16 +121,15 @@ def do_resolve_with_resolved_children(
             # overwrite the SnowflakeCreateTable resolving, because the child
             # attribute will be pulled directly from the cache
             resolved_child = resolved_children[logical_plan.children[0]]
-            # when the plan is for SnowflakeCreateTable, there must be a snowflake_create_table_plan_info
-            # associated with the current query generator.
-            # TODO: this check need to be relaxed when large query breakdown with temp table
-            #       is implemented, because the child attributes are not necessary to create
-            #       the temp table
-            assert self._snowflake_create_table_plan_info is not None
-            assert (
-                self._snowflake_create_table_plan_info.table_name
+            child_attributes = None
+            if (
+                self._snowflake_create_table_plan_info
```

Comment:
let's be more strict about this part of code, since we have the logical_plan.creation_source here, let's do a check that if the creation_source is large query breakdown, set the child attribute to None, else, we can keep the original assertion and get the child attribute from the map.

Reply:
@sfc-gh-aalam Can we keep the original assertion check when the create source is not large query breakdown?